core/frontend/src/WebViewerApp.ts

```diff
@@ -51,8 +51,12 @@ export class WebViewerApp {
       const redirectUrl = opts.webViewerApp.authConfig.redirectUri;
       const urlObj = new URL(redirectUrl);
       if (urlObj.pathname === window.location.pathname) {
-        await BrowserAuthorizationCallbackHandler.handleSigninCallback(redirectUrl);
-        return;
+        try {
+          await BrowserAuthorizationCallbackHandler.handleSigninCallback(redirectUrl);
+          return;
+        } catch (err) {
+          Logger.logTrace(loggerCategory, "Unable to handle signin callback", () => ({ message: err.toString() }));

```

Comment:
I don't understand why `handleSigninCallback` throws an exception. Please explain.

But, if you're going to "mask" an exception, please use `UnexpectedErrors.handle` (for 3.0).

Reply:
Hmm, looks to me that `BrowserAuthorizationCallbackHandler.handleSigninCallback` could very well be made to no longer throw exceptions and simply log the callback errors instead.

Nearly all situations I can imagine surrounding a failed callback in a popup or iframe will close the child window anyway, so unless something goes _really_ wrong (which would likely indicate an issue with the auth provider itself), we can suppress the errors within the function, instead of making that the responsibility of `WebViewerApp`

@ramanujam-raman do you have any objections to doing this?

Reply:
I've removed the source of the thrown error within `BrowserAuthorizationCallbackHandler.handleSigninCallback` and reverted the WebViewerApp's behavior back to the way it was before. This way, _all_ callback errors are suppressed.

I should note that the errors being thrown here aren't always expected, though. Errors occurring during callback handling could indicate that a legitimate issue occurred during signin; they're not always white noise.

The errors themselves are indirectly causing the issue raised in the PR, which is related to the inability to use the same URL for app startup _and_ signin callbacks. When these URLs are different, callback handling only occurs when the callback URL is visited, but when they're the same, callback handling needs to happen every time, which means callback handling errors can happen because of a login error _or_ because there wasn't a callback at all. Errors thrown in the former case are unexpected, while errors thrown in the latter case are meaningless (and can be ignored).

Logins via redirect can make use of errors during callback handling to navigate into contingency plans or display clean error messages to the user.
Suppressing all errors thrown by `handleSigninCallback` removes these approaches while resolving the above issue.

If that seems like an acceptable tradeoff, I'll move to port these changes into 3.x as well.
Given the above description, though, Should we still report the errors using `UnexpectedErrors`?

Reply:
for cases where the errors are expected and meaningless, just ignore them (log if you think someone might want to know they happened). If they're unexpected and unrecoverable, yes for 3.0 use `UnexpectedErrors`. That will cause a fatal error in development, and log a fatal error (also notifying fatal-error-listeners) and then swallow it in production. Of course that presumes the program can continue after such an error.